### PR TITLE
add sample webapp and resource in the same application

### PR DESCRIPTION
**Pull request by moarychan**
Fixes Azure/azure-sdk-for-java#19816

Related prerequisites PR: [Support web app and web api in the same application](https://github.com/Azure/azure-sdk-for-java/pull/22674
)
Changelog:
- Add a new sample to apply web app and web api in the same scenario.
- Use the new value `on_behalf_of` for readme and YAML files.
- Add `appRoles` setting description for client credentials scope.

### CODE REVIEW

**Comment by moarychan** [comment]
Hi @chenrujun , please help to review together.

File: aad/azure-spring-boot-sample-active-directory-webapp-and-resource-server/pom.xml [hunk 1]
```diff
@@ -0,0 +1,59 @@
...
+        <version>1.17.0</version> <!-- {x-version-update;com.azure:azure-core;dependency} -->
+      </dependency>
+    </dependencies>
+  </dependencyManagement>
+
+  <dependencies>
+    <dependency>
+      <groupId>com.azure.spring</groupId>
+      <artifactId>azure-spring-boot-starter-active-directory</artifactId>
+      <version>3.7.0-beta.1</version> <!-- {x-version-update;com.azure.spring:azure-spring-boot-starter-active-directory;current} -->
```
Comment (backwind1233) [inline aad/azure-spring-boot-sample-active-directory-webapp-and-resource-server/pom.xml:35]: if the branch is in development ,it is ok.
but if you want to merge this branch to main branch,
we should not depend on unreleased lib on main branch.
the github action(pipeline) will not pass

Comment (moarychan) [reply aad/azure-spring-boot-sample-active-directory-webapp-and-resource-server/pom.xml:35]: I got it.

File: aad/azure-spring-boot-sample-active-directory-resource-server-obo/README.md [hunk 2]
```diff
@@ -60,7 +60,7 @@ azure:
         scopes:
           - https://graph.microsoft.com/User.Read
       webapiB:         # When authorization-grant-type is null, on behalf of flow is used by default
-        authorization-grant-type: on-behalf-of
+        authorization-grant-type: on_behalf_of
```
Comment (backwind1233) [inline aad/azure-spring-boot-sample-active-directory-resource-server-obo/README.md:63]: will "on-behalf-of" also work?

**Comment by backwind1233** [comment]
@chenrujun please help review this sample

File: aad/azure-spring-boot-sample-active-directory-resource-server-obo/README.md (hunk 2 above)
Comment (moarychan) [reply aad/azure-spring-boot-sample-active-directory-resource-server-obo/README.md:63]: Yes.

**Review by chenrujun — APPROVED** [review]
LGTM.

**Pull request merged**